src/UserRepository/HtpasswdFactory.php

```diff
@@ -24,9 +24,10 @@ class HtpasswdFactory
      */
     public function __invoke(ContainerInterface $container): Htpasswd
     {
-        $config = $container->get('config');
+        $config = (array) $container->get('config');
```

Comment:
This might not do what is expected unless it is ArrayObject or `stdObject`.
For stdlib config object we would want to call `toArray();` and for iterable we should probably do `iterator_to_array()`, otherwise reject altogether

Reply:
`stdClass` or `object` won't allow array access and thus would've triggered massive notices in the past, thus I dont really care about that.

With `stdlib config object` you mean `laminas-config`? As this has nothing todo with `laminas-stdlib`, I am a bit confused.

Reply:
`iterator_to_array` would only work if the `$config` is an instance of `IteratorAggregate` but thats not the case if the provided object only provides `ArrayAccess`...

Reply:
Yes, i meant laminas-config.
We actually require 'config' to be an array, which poses an issue. It is far reaching issue that I will bring to contributors channel.